### PR TITLE
Add levenshtein from polyleven

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Add a simple levenshtein distance function using the implementation from the `polyleven` library as `spacy.matcher.levenshtein`.

### Types of change

Enhancement.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
This is initially a sketch of what this could look like for use with a `FUZZY` `Matcher` operator: #11359.

**Review by polm — COMMENTED** [review]
This looks good, I was able to run it locally without issue. I also ran a benchmark and noticed it was slightly faster in spaCy, similar to polyleven with k=3. Not sure why that would happen.

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,7 @@ quickstart-training-generator.js
 cythonize.json
 spacy/*.html
 *.cpp
+*.c
```
[the remark this answers is not in the document]
Comment (adrianeboyd) [reply .gitignore:27]: This generates `levenshtein.c` from cython.

**Comment by adrianeboyd** [comment]
My guess is that the `cpdef` interface is somehow slightly faster than the standard python one in `polyleven`, but I don't actually know for sure.

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (polm) [reply .gitignore:27]: Right, I missed that. That's fine then.

**Review by svlandeg — APPROVED** [review]
Looks good!

**Pull request merged**